web_monitoring/db.py

```diff
@@ -1,47 +1,657 @@
+    return f'{start_str}..{end_str}'
+
+
+def _build_version(*, page_id, uuid, capture_time, uri, hash, source_type, title,
+                   source_metadata=None):
+    """
+    Build a Version dict from parameters, performing some validation.
+    """
+    if not isinstance(capture_time, str):
+        capture_time = _tzaware_isoformat(capture_time)
+    if source_metadata is None:
+        source_metadata = {}
+    version = {'page_id': page_id,
+               'uuid': uuid,
+               'capture_time': capture_time,
+               'uri': str(uri),
+               'hash': str(hash),
+               'source_type': str(source_type),
+               'title': str(title),
+               'source_metadata': source_metadata}
+    return version
+
+
+def _build_importable_version(*, page_url, uuid, capture_time, uri, hash,
+                             source_type, title, source_metadata=None):
     """
-    # Existing documentation of import API is in this PR:
-    # https://github.com/edgi-govdata-archiving/web-monitoring-db/pull/32
-    url = WEB_MONITORING_CREATE_IMPORT_API.format(db_url=settings['db_url'])
-    return requests.post(url,
-                         auth=(settings['db_email'], settings['db_password']),
-                         headers={'Content-Type': 'application/x-json-stream'},
-                         data='\n'.join(map(json.dumps, versions)))
+    Build a Version dict from parameters, performing some validation.
+
+    This is different than _build_version because it needs ``page_url`` instead
+    of ``page_id`` of an existing Page.
+    """
+    if not isinstance(capture_time, str):
+        capture_time = _tzaware_isoformat(capture_time)
+    if source_metadata is None:
+        source_metadata = {}
+    version = {'page_url': page_url,
+               'uuid': uuid,
+               'capture_time': capture_time,
+               'uri': str(uri),
+               'hash': str(hash),
+               'source_type': str(source_type),
+               'title': str(title),
+               'source_metadata': source_metadata}
+    return version
+
+class MissingCredentials(RuntimeError):
+    ...
 
 
-def query_import_status(import_id):
+class Client:
     """
-    Check on the status of a bulk import job.
+    Communicate with web-monitoring-db via its REST API.
+
+    This object encapsulates authentication information and provides
+    methods corresponding to the REST API.
+
+    The Client can also be configured via environment variables using the
+    class method :meth:`Client.from_env`.
 
     Parameters
     ----------
-    import_id : integer
+    url : string
+    email : string
+    password : string
     """
-    url = WEB_MONITORING_SHOW_IMPORT_API.format(db_url=settings['db_url'],
-                                                import_id=import_id)
-    return requests.get(url,
-                        auth=(settings['db_email'], settings['db_password']))
+    def __init__(self, url, email, password):
```

Comment:
Should `url` have a default (`https://api.monitoring.envirodatagov.org`)? I wouldn’t expect one for `email` and `password`, but I probably would for the API host.